### PR TITLE
Correct the actual allocated size from allocator when call sdsRedize to align the logic with sdsnewlen function.

### PR DESCRIPTION
**Pull request by zhulipeng**
This patch try to correct the actual allocated size from allocator
when call sdsRedize to align the logic with sdsnewlen function.

Maybe the https://github.com/valkey-io/valkey/pull/453 optimization should depend on this.

### CODE REVIEW

**Comment by poiuj** [comment]
Right, I did very similar changes. Didn't push yet.

File: src/sds.c [hunk 1]
```diff
@@ -1470,27 +1472,27 @@ int sdsTest(int argc, char **argv, int flags) {
         x = sdsResize(x, 200, 1);
         test_cond("sdsrezie() expand len", sdslen(x) == 40);
         test_cond("sdsrezie() expand strlen", strlen(x) == 40);
-        test_cond("sdsrezie() expand alloc", sdsalloc(x) == 200);
+        test_cond("sdsrezie() expand alloc", sdsalloc(x) >= 200);
```
Comment (madolson) [inline src/sds.c]: These need to get ported to the new tests, but can we also add some reasonable upper bound like 
```suggestion
        test_cond("sdsrezie() expand alloc", sdsalloc(x) >= 200 && sdsalloc(x) < 400);
```
Also document at least once why we are doing the bounds checking.

**Review by madolson — COMMENTED** [review]
This looks good to me, can you move the tests to the new framework?

**Comment by zhulipeng** [comment]
> This looks good to me, can you move the tests to the new framework?

Done.

**Review by madolson — APPROVED** [review]
A nice to have, but not necessarily in the scope of this PR, is we should probably add more checks that `sdsalloc()` is actually always equal to the underlying zmalloc allocation size.

**Comment by poiuj** [comment]
Just to clarify, underlying allocation size equals to `sdsalloc()` + header size + null terminator. `sdsAllocSize()` should calculate it.

**Comment by madolson** [comment]
@lipzhu A test started to fail after we merged this change related to allocated size: 
https://github.com/valkey-io/valkey/actions/runs/9143545732/job/25140329542#step:6:1505
https://github.com/valkey-io/valkey/actions/runs/9135654696/job/25123330670#step:6:1501

It's the libc malloc function, let me know if you think it's related and if you can try to fix it.

**Comment by zhulipeng** [comment]
@madolson Sure, I will take a look and try to reproduce in local.

**Pull request merged**